homeassistant/components/ecovacs/__init__.py

```diff
@@ -38,32 +37,46 @@
     extra=vol.ALLOW_EXTRA,
 )
 
-ECOVACS_DEVICES = "ecovacs_devices"
-
-# Generate a random device ID on each bootup
-ECOVACS_API_DEVICEID = "".join(
-    random.choice(string.ascii_uppercase + string.digits) for _ in range(8)
-)
+PLATFORMS = [
+    Platform.VACUUM,
+]
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Ecovacs component."""
-    _LOGGER.debug("Creating new Ecovacs component")
+    if DOMAIN in config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
 
     def get_devices() -> list[VacBot]:
-        ecovacs_api = EcoVacsAPI(
-            ECOVACS_API_DEVICEID,
-            config[DOMAIN].get(CONF_USERNAME),
-            EcoVacsAPI.md5(config[DOMAIN].get(CONF_PASSWORD)),
-            config[DOMAIN].get(CONF_COUNTRY),
-            config[DOMAIN].get(CONF_CONTINENT),
-        )
-        ecovacs_devices = ecovacs_api.devices()
-        _LOGGER.debug("Ecobot devices: %s", ecovacs_devices)
+        try:
+            ecovacs_api = EcoVacsAPI(
+                get_client_device_id(),
+                entry.data[CONF_USERNAME],
+                EcoVacsAPI.md5(entry.data[CONF_PASSWORD]),
+                entry.data[CONF_COUNTRY],
+                entry.data[CONF_CONTINENT],
+            )
+            ecovacs_devices = ecovacs_api.devices()
+        except ValueError as ex:
+            _LOGGER.error("Ecovacs login failed due to wrong username/password")
+            raise ConfigEntryError("invalid username or password") from ex
+        except RuntimeError as ex:
+            _LOGGER.exception("Unexpected exception")
```

Comment:
Normally we don't log when we try again to avoid spamming the log. We can pass the message as an exception argument, which will be logged.

We also normally don't catch unknown exceptions outside of the config flow.

Reply:
Will add the error handling in a follow-up including the new lib